content/blog/2021-12-02-collaborative-experiments.md

```diff
@@ -0,0 +1,151 @@
+and you want to get their workspace ready quickly. Or you might need to change
+computers to access better resources to run experiments on.
+
+Setting up DVC remotes in addition to your Git remotes lets you share all of the
+data, code, and hyperparameters associated with each experiment so anyone can
+pick up where you left off in the training process. We'll go through an example
+of sharing an experiment with DVC remotes.
+
+## Getting the project
+
+To follow along, you can fork
+[this repo](https://github.com/iterative/example-dvc-experiments) as one of your
+own GitHub repos. That way you'll have pull access when we start working with
+DVC. This repo has a few different branches that have experiments associated
+with them and you're welcome to check them out!
+
+To get the branch we'll use in this post, you can run this command.
+
+```dvc
+$ git clone https://github.com/iterative/example-dvc-experiments/ -b get-started
+```
+
+This project already has DVC files set up to run experiments, but if you want to
+follow along with a project you're currently working on, make sure to check out
+the steps to initialize a DVC pipeline in
+[the Getting Started doc](https://dvc.org/doc/start).
+
+## Setting up your DVC remotes
+
+When you want to share the progress you've made with training your model, that
+usually means you need to find a way to bundle the code, data, and
+hyperparameters. This could be a complicated process if you're working with GBs
+worth of data or you have a large number of hyperparameters.
+
+That's one of the uses for DVC and why we'll be working with remotes. To start
+with, make sure your GitHub remote is configured correctly. It should use the
+SSH version of the URL. This is so DVC can authenticate the pushes and pulls
+from GitHub it needs as part of experiment sharing.
+
+The way DVC works is by storing custom Git refs in your repo with metadata that
+defines the experiment. You can learn more about how DVC uses custom Git refs in
+[this post](https://dvc.org/blog/experiment-refs).
+
+Next, you'll need to set up a remote to your data location. This could be an AWS
+S3 bucket, a Google Drive, or
+[one of the other supported storage types](https://dvc.org/doc/command-reference/remote/add#supported-storage-types).
+For this example, we'll be using a Google Drive folder as the remote to handle
+pushing data.
+
+An important thing to note about the project we're working with is that there is
+already a remote set up for you to pull from. You can see this in `.dvc/config`.
+You'll need to set up a separate remote to push changes to since this remote
+doesn't allow push access.
+
+Now that you know what we're doing, let's run the command to set up our DVC
+remote.
+
+```dvc
+$ dvc remote add cloud_remote gdrive://1k6aUYWphOulJlXgq4XbfKExWGyymTpEl
+```
+
+This adds the remote storage named `cloud_remote` for DVC to track and we'll be
+able to push and pull the exact code and data to reproduce any experiment from
```

Comment:
To push the _exact code_ (and hyperparams), the reader should have write permission to the _Git_ remote. Even if they use SSH URLs, if they don't have the necessary permissions, they can't push the experiments. 

The flow, IMO, could be like: 

- Asking the user to _fork_ `example-dvc-experiments` to their Github account
- Clone this forked repo to their machine
- Add a _DVC remote_, e.g. Google Drive
- Add a Git remote called _upstream_ to `https://github.com/iterative/example-dvc-experiments` 
- Now:

```
dvc exp list upstream 
```

will show the experiments in **our** repository. They can pull these experiments like:

```
dvc exp pull upstream cnn-128
```

After pulling these experiments, they can push to their repository,

```
dvc exp push origin cnn-128
```

This way, we show how to share experiments. 

Actually, you don't have to specify DVC remote, for just _read-only_ access to the models, data, etc., they can use our DVC remote. They can't push their experiment artifacts to ours, but that's a side issue here I think. 

The tagline could be: "If you and your colleagues have Github accounts, you can do collaborative experiments"